pandas/tests/indexes/test_common.py

```diff
@@ -391,3 +400,61 @@ def test_astype_preserves_name(self, index, dtype):
             assert result.names == index.names
         else:
             assert result.name == index.name
+
+
+@pytest.fixture(
+    params=[
+        tm.makeBoolIndex(10),
+        tm.makeFloatIndex(10),
+        tm.makeIntervalIndex(10),
+        tm.makeStringIndex(10),
+        tm.makeIntIndex(10),
+        tm.makeUIntIndex(10),
+        tm.makeRangeIndex(10),
+        tm.makeCategoricalIndex(10),
+        tm.makeMultiIndex(10),
+        tm.makeDateIndex(10),
+        tm.makePeriodIndex(10),
+        tm.makeTimedeltaIndex(10),
+    ]
+)
+def index_with_missing(request):
+    ind = request.param
+    if type(ind) in [DatetimeIndex, PeriodIndex, TimedeltaIndex]:
+        pytest.xfail("stable descending order sort not implemented")
+    if type(ind) in [Int64Index, UInt64Index, RangeIndex]:
+        pytest.xfail("missing values not supported")
+    if type(ind) in [CategoricalIndex, MultiIndex]:
+        pytest.xfail("missing value sorting order not defined for index type")
+    vals = request.param.values
+    vals[2] = None
+    vals[5] = None
+    ind = type(ind)(vals)
+    return ind
+
+
+@pytest.mark.parametrize("na_position", ["middle"])
+def test_sort_values_invalid_na_position(index_with_missing, na_position):
+    if na_position not in ["first", "last"]:
+        with pytest.raises(
+            ValueError, match=f"invalid na_position: {na_position}",
+        ):
+            index_with_missing.sort_values(na_position=na_position)
+
+
+@pytest.mark.parametrize("na_position", ["first", "last"])
```

Comment:
is `na_position=None` valid? (it *should* just default or error if we don't accept it), so test either here or above if its supposed to error.

Reply:
`na_position=None` isn't valid. Added a test case.